---
title: "LinDA-example"
author: "Elsa"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### LinDA
#### phylum level
```{r LinDA DA P, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq.phylum <- tax_glom(physeq, 'phylum')
tse = mia::makeTreeSummarizedExperimentFromPhyloseq(physeq.phylum) # create tse object from your phyloseq object

tse_LinDA <- mia::subsetByPrevalentTaxa(tse, detection = 0, prevalence = 0.1) # filter tse object if needed
```
```{r LinDA DA P analysis, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Un-comment lines below to install LinDA from GitHub if needed

# install.packages(c("modeest", "lmerTest", "foreach", "parallel", "ggplot2", "ggrepel"))
# install.packages("devtools")
# devtools::install_github("zhouhj1994/LinDA")

otu.tab <- as.data.frame(assay(tse_LinDA))
meta <- as.data.frame(colData(tse_LinDA)) %>% select(region)
LinDA_phylum <- LinDA::linda(
  otu.tab, 
  meta, 
  formula = '~region', 
  type = "count", # options are type and proportion. defines the otu table data type. default is count
  adaptive = TRUE, #default is TRUE. tells the function to choose the significance test most appropriate based on sequencing depth
  imputation = FALSE, # can only be set if adaptive = FALSE. default is FALSE. if true, zeros in otu table are imputed. 
  pseudo.cnt = 0.5, # default is 0.5. can only be implemented if imputation and adaptive are both FALSE
  p.adj.method = 'BH', #default is BH, can be set to any of the options in p.adjust
  alpha = 0.05, # significance level, default is 0.05
  prev.cut = 0, # taxa with a prev.cut (prevalence -- percentage of non-zero values) lower     than the selected threshold are excluded. input values are 0-1
  lib.cut = 1000, # taxa with a lib.cut lower than the selected threshold are excluded.        default is 1, input is any non-negative value
  winsor.quan = 0.97, # sets the winsorization threshold. Default is NULL, meaning all values are included in analysis. In the example data with 0.97, 3% of otu table data at the extremes are removed from the analysis (limits extremes and outliers)
  n.cores = 1, # default is 1. any positive integer, can be applied or changed in the case of a mixed-effect model
  )

# ?linda is very helpful! 

LinDA_phylum
LinDA::linda.plot(LinDA_phylum, c('region') )

LinDA_table_phylum <- rownames_to_column(LinDA_phylum$output$regionIttoqqortoormii, "phylum") 
```
```{r LinDA DA P summary, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq@tax_table) %>%
  rownames_to_column(., "taxon")

LinDA_table_phylum <- rownames_to_column(LinDA_phylum$output$regionIttoqqortoormii, "genome") 
LinDA_table_phylum <- merge(LinDA_table_phylum, taxonomyclean[, c("genome", "phylum")], by = "genome", all.x = TRUE)

LinDA_table_phylum <- LinDA_table_phylum %>%
  dplyr::select(phylum, lfcSE, padj, log2FoldChange) %>%
  filter(padj < 0.005)

colors_alphabetic <- ehi_phylum_colors1 %>%
  dplyr::right_join(taxonomyclean, by=join_by(phylum == phylum)) %>%
  select(phylum, colors) %>%
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(LinDA_table_phylum$phylum))
colnames(tax_table)[1] <- "phylum"
# tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
#	dplyr::arrange(phylum) %>%
#	select(colors) %>%
#	pull()
```
```{r LinDA DA P plotting, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ggplot(LinDA_table_phylum, aes(x=forcats::fct_rev(phylum), y=log2FoldChange, color=phylum)) + 
  geom_point(size=4) + 
  #scale_color_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  xlab("phylum") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))
```
