---
title: "Elevation Hologenomics v2"
author: "Garazi Martin Bideguren"
date: "2023-12-06"
output: 
  html_document:
    keep_md: yes
    theme: simplex
    toc: yes
    toc_float:
      collapsed: false
    fig_width: 7 
    fig_height: 4
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Experiment information

The samples correspond to the four transects from the Spanish and French side of the Pyrenees.\
Per transect we have 4-6 sites, running from 900m to 2400 ASL and per sampling point we have 4-5 individuals sampled.\
Coassemblies had been formed per elevation, having 10 coassembly groups per country and running from the lowest elevation to the highest.\
The MAGs have been dereplicated all together.\

| **Country** | **Transect** | **Site** | **Elevation** |
|:-----------:|:------------:|:--------:|:-------------:|
|    Spain    |     Aisa     |    A     |     1250m     |
|             |              |    B     |     1450m     |
|             |              |    C     |     1650m     |
|             |              |    D     |     1850m     |
|             |     Aran     |    E     |     1000m     |
|             |              |    F     |     1080m     |
|             |              |    G     |     1340m     |
|             |              |    H     |     1500m     |
|             |              |    I     |     1650m     |
|             |              |    J     |     1850m     |
|   France    |  Tourmalet   |    K     |     941m      |
|             |              |    L     |     1260m     |
|             |              |    K     |     1628m     |
|             |              |    N     |     1873m     |
|             |   Sentein    |    O     |     953m      |
|             |              |    P     |     1255m     |
|             |              |    Q     |     1561m     |
|             |              |    R     |     1797m     |
|             |              |    S     |     2100m     |
|             |              |    T     |     2306m     |

![**Map: sampling sites**](/Users/dlz554/Desktop/PhD_project/Elevation_Hologenomics/Rmd/Map.png "Map")

## Data preparation

### Load required libraries

```{r libraries, message=FALSE, results=FALSE}
library(tidyverse)
library(ape)
library(devtools)
library(ggplot2)
library(ggpubr)
library(hilldiv2)
library (knitr)
library(spaa)
library(vegan)
library(lme4)
```

## Declare directories and files

```{r directories}
workingdir="/Users/dlz554/Desktop/PhD_project/Elevation_Hologenomics/Rmd/Elevatin_Hologenomics_v2"

counts_file="DMB0113_counts.tsv"
tree_file="DMB0113.tree"
taxonomy_file="DMB0113_mag_info.tsv"
metadata_file="Pyrenees_metadata_all.tsv"
coverage_file="DMB0113_coverage.tsv"
kegg_file="GIFTs.tsv"
```

## Load data

```{r loaddata}
ehi_phylum_colors <- read.table("ehi_phylum_colors.tsv",sep="\t",header=T,comment.char = "")

count_table <- read.table(counts_file,header=T, sep="\t",row.names=1)
tree <- read.tree(tree_file)
mags_table <- read.table(taxonomy_file,sep="\t",header=T)
rownames(mags_table) <- mags_table[,1] # add row names
metadata <- read.table(metadata_file,header=T, sep=",")
coverage_table <- read.table(coverage_file,header=T, sep="\t", row.names=1)
kegg_table <- read.table(kegg_file,header=T, sep="\t", row.names=1)
```

## Filtering

After the EHI report we saw that EHI00102 should be excluded for excessive host DNA, and EHI00182 due to excessive not known content from the Spanish Transects. EHI00435 had 0 of bins and 0% of assembly mapping so was also excluded from the French Transect.

```{r filterging, echo=TRUE, warning=FALSE}

#Counts_raw
columns_to_exclude <- c("EHI00102", "EHI00182","EHI00435") # Columns to exclude
count_table <- count_table %>%
  select(-columns_to_exclude)

#Metadata
metadata <- metadata %>%
  filter(EHI_number != "EHI00102") %>%
  filter(EHI_number != "EHI00182") %>%
  filter(EHI_number !="EHI00435")

#Coverage_table
columns_to_exclude <- c("EHI00102", "EHI00182","EHI00435")  # Columns to exclude
coverage_table <- coverage_table %>%
  select(-columns_to_exclude)
```

## Sequencing depth

```{r seq, echo=TRUE, warning=FALSE}
sequencing_depth <- colSums(count_table)
```

## MAG catalogue

```{r mag_quality_table}
mag_details <- mags_table %>%
  select(c(genome,domain,phylum,completeness,contamination,mag_size)) %>%
  mutate(mag_size=round(mag_size/1000000,2)) %>% #change mag_size to MBs
  rename(comp=completeness,cont=contamination,size=mag_size) %>% #rename columns
  remove_rownames() %>%
  arrange(match(genome, rev(tree$tip.label))) #sort MAGs according to phylogenetic tree
```

## DNA fractions

```{r frac, echo=TRUE, warning=FALSE}
sequence_fractions <- count_table %>%
  rownames_to_column("Genome") %>%
  pivot_longer(-Genome, names_to = "sample", values_to = "value") %>%
  group_by(sample) %>%
  summarise(mags = sum(value)) %>%
    left_join(metadata, by = join_by(sample == EHI_number))  %>%
    select(sample,mags,metagenomic_bases,host_bases,bases_lost_fastp_percent) %>%
    mutate(mags_bases = mags*146) %>%
    mutate(lowqual_bases = ((metagenomic_bases+host_bases)/(1-bases_lost_fastp_percent))-(metagenomic_bases+host_bases)) %>%
    mutate(unmapped_bases = metagenomic_bases - mags_bases) %>%
    mutate(unmapped_bases = ifelse(unmapped_bases < 0, 0, unmapped_bases)) %>%
    select(sample,mags_bases,unmapped_bases,host_bases,lowqual_bases)

mags_bases_mean <- sequence_fractions %>%
    mutate(mags_bases = mags_bases / 1000000000) %>%
    select(mags_bases) %>%
    pull() %>%
    mean()
```


## Count data

### Minimum genome-coverage filtering

```{r coverage_filtering}
min_coverage=0.3
count_table_cov <- coverage_table %>%
  mutate(across(everything(), ~ ifelse(. > min_coverage, 1, 0))) %>%
  map2_df(., count_table, ~ .x * .y) %>%
  as.data.frame()
rownames(count_table_cov) <- rownames(coverage_table)
```

### Genome size normalisation

```{r genome_size_normalisation}
genome_read_sizes <- mags_table[rownames(count_table_cov),] %>%
    select(mag_size) %>%
    mutate(mag_size = mag_size / 150) %>%
    pull()
count_table_cov_size <- sweep(count_table_cov, 1, genome_read_sizes, "/")
```

## Community barplots

The count data table requires some modifications before generating the plots, including TSS normalisation, appending taxonomy and metadata information etc. The following script generates a tibble that can then be used to plot identical stacked barplots coloured at different taxonomic levels.

```{r taxonomy_table, warning=FALSE, comments="", message=FALSE}
count_table_cov_size_pivot <- count_table_cov_size %>%
  rownames_to_column("Genome") %>%
  mutate_at(vars(-Genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-Genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., mags_table, by = join_by(Genome == genome)) %>% #append taxonomy
  mutate(phylum = fct_relevel(phylum, rev(ehi_phylum_colors$phylum))) #sort phyla by taxonomy
```

```{r taxonomy_plot, warning=FALSE, comments="", message=FALSE}
# Retrieve taxonomy colors to use standardised EHI colors
phylum_colors <- ehi_phylum_colors %>%
  filter(phylum %in% unique(count_table_cov_size_pivot$phylum)) %>%
  select(colors) %>%
  pull() %>%
  rev()
phylum_colors <- c(phylum_colors,"#cccccc") #REMOVE! ONLY FOR ARCHAEANS

count_table_cov_size_met <- count_table_cov_size %>%
  rownames_to_column("Genome") %>%
  mutate_at(vars(-Genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-Genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., mags_table, by = join_by(Genome == genome)) %>% #append taxonomy
  left_join(., metadata, by = join_by(sample == EHI_number)) %>%
  mutate(phylum = fct_relevel(phylum, rev(ehi_phylum_colors$phylum))) #sort phyla by taxonomy


# Plot stacked barplot
ggplot(count_table_cov_size_met, aes(x=sample,y=count,fill=phylum, group=phylum))+ #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1)+ #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    labs(y = "Relative abundance") +
    guides(fill = guide_legend(ncol = 3)) +
    facet_grid(. ~ country)+
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
					axis.title.x = element_blank(),
          panel.background = element_blank(),
          panel.border = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          legend.position="top",
          legend.title=element_blank())
```

# Diversity analysis

## Data preparation

In order to avoid issues with diversity computation is recommendable to remove samples and MAGs without count data.

```{r diversity_filtering}
#Get list of present MAGs
present_MAGs <- count_table_cov_size %>%
		filter(rowSums(.[, -1]) != 0) %>%
		rownames()

#Remove samples with all zeros (no data after filtering)
count_table_cov_size <- count_table_cov_size %>%
  select_if(~!all(. == 0))

#Align KEGG annotations with present MAGs and remove all-zero and all-one traits
present_MAGs <- present_MAGs[present_MAGs %in% rownames(kegg_table)]
kegg_table_filt <- kegg_table[present_MAGs,] %>%
			select_if(~!all(. == 0)) %>%  #remove all-zero modules
			select_if(~!all(. == 1)) #remove all-one modules

#Filter count table to only contain present MAGs after KEGG filtering
count_table_cov_size_filt <- count_table_cov_size[present_MAGs,]
```

## Alpha diversity

### Alpha diversity metrics{-}

```{r alpha_diversity, warning=FALSE, comments="", message=FALSE}
q0n <- hilldiv(count_table_cov_size,q=0) %>% c()
q1n <- hilldiv(count_table_cov_size,q=1) %>% c()
q1p <- hilldiv(count_table_cov_size,q=1,tree=tree) %>% c()
dist <- traits2dist(kegg_table_filt, method="gower")
q1f <- hilldiv(count_table_cov_size_filt,q=1,dist=dist) %>% c()

# Merge all metrics
alpha_div <- cbind(sample=colnames(count_table_cov_size),richness=q0n,neutral=round(q1n,3),phylo=round(q1p,3),func=round(q1f,3)) %>%
  as.data.frame()
columns <- c("richness","neutral","phylo","func","mapped","total")

# Add amount of sequencing data to the table
alpha_div <- alpha_div %>%
  left_join(sequence_fractions, by = join_by(sample == sample)) %>% #add sequencing depth information
  mutate(mapped=round(mags_bases/1000000000,3)) %>% #modify depth to million reads
  mutate(total=round((mags_bases+unmapped_bases+host_bases+lowqual_bases)/1000000000,3)) %>%
  select(sample,richness,neutral,phylo,func,mapped,total) %>%
  mutate(across(-1, as.numeric))
```

### Alpha diversity plots{-}

```{r alpha_diversity_plot, fig.height=8}
alpha_div %>%
		pivot_longer(-sample, names_to = "data", values_to = "value") %>%
		mutate(data = factor(data, levels = columns))	%>%
		ggplot(aes(x=value, y=sample)) +
			geom_bar(stat='identity', fill="#6c9ebc") +
			facet_wrap(~data,  scales="free_x", ncol=6) +
			theme_classic() +
			theme(
				strip.background = element_blank(),
				panel.grid.minor.x = element_line( size=.1, color="grey" ),
				axis.title.x = element_blank(),
				axis.title.y = element_blank(),
				axis.text.x = element_text(angle = 45, hjust = 1)
			)
```

You can also generate an HTML table using knitr.

```{r alpha_diversity_table}
kable(alpha_div)
```


### Alpha diversity comparisons{-}
Alpha diversities can be compared across any categorical features that group analysed samples (e.g., localities, sampling seasons, sex), or continuous variables associated with the host animals.

Let's first create a nice colour palette for the localities
```{r alpha_colors}
alpha_colors <- c("#e5bd5b","#6b7398","#76b183","#d57d2c","#2a2d26","#f9d4cc","#3c634e","#ea68c3")
```

Let's also identify the number of comparing groups, so that the colour palette can be subsetted properly when plotting the figures.
```{r alpha_divs_comp_groups}
group_n <- alpha_div %>% select(sample,neutral) %>%
		left_join(metadata, by = join_by(sample == EHI_number)) %>%
		mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
		select(location) %>% pull() %>% unique() %>% length()
```

#### Neutral diversity{-}
```{r alpha_divs_comp_neutral}
alpha_div %>%
			select(sample,neutral) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(metadata, by = join_by(sample == EHI_number)) %>%
			mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
			ggboxplot(., x = "location", y = "value", color = "Transect") +
					scale_color_manual(values=alpha_colors[c(1:group_n)]) +
					scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)])) +
					stat_compare_means() +
					theme_classic() +
					labs(y = "Neutral Hill numbers") +
					theme(
						legend.position = "top",
						legend.box = "horizontal",
						axis.title.x = element_blank(),
						axis.text.x = element_blank()) +
					guides(color=guide_legend(title="Location"))
```

#### Phylogenetic diversity{-}
```{r alpha_divs_comp_phylo}
alpha_div %>%
			select(sample,phylo) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(metadata, by = join_by(sample == EHI_number)) %>%
			mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
			ggboxplot(., x = "location", y = "value", color = "Transect") +
			scale_color_manual(values=alpha_colors[c(1:group_n)]) +
			scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)])) +
			stat_compare_means() +
			theme_classic() +
			labs(y = "Phylogenetic Hill numbers") +
			theme(
						legend.position = "top",
						legend.box = "horizontal",
						axis.title.x = element_blank(),
						axis.text.x = element_blank()) +
					guides(color=guide_legend(title="Location"))
```

#### Functional diversity{-}
```{r alpha_divs_comp_func}
alpha_div %>%
			select(sample,func) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(metadata, by = join_by(sample == EHI_number)) %>%
			mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
			ggboxplot(., x = "location", y = "value", color = "Transect") +
			scale_color_manual(values=alpha_colors[c(1:group_n)]) +
			scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)])) +
			stat_compare_means() +
			theme_classic() +
			labs(y = "Functional Hill numbers") +
			theme(
						legend.position = "top",
						legend.box = "horizontal",
						axis.title.x = element_blank(),
						axis.text.x = element_blank()) +
					guides(color=guide_legend(title="Location"))
```


### Regression plot

#### Neutral diversity

```{r alpha_div_neutral_plot, echo=TRUE, message=FALSE, warning=FALSE}

alpha_div_neutral_met<-alpha_div %>%
			select(sample,neutral) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(metadata, by = join_by(sample == EHI_number))


ggplot(alpha_div_neutral_met, aes(x = value, y = Elevation)) +
  geom_point() +
  geom_smooth(method = lm) +
  facet_wrap(~ factor(Transect))+
  labs(x = "Alpha div. neutral")
```

#### Phylogenetic diversity

```{r alpha_div_phylo_plot, echo=TRUE, message=FALSE, warning=FALSE}
alpha_div_phylo_met<-alpha_div %>%
			select(sample,phylo) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(metadata, by = join_by(sample == EHI_number))


ggplot(alpha_div_phylo_met, aes(x = value, y = Elevation)) +
  geom_point() +
  geom_smooth(method = lm) +
  facet_wrap(~ factor(Transect))+
  labs(x = "Alpha div. phylogenetic")
```

#### Functional diversities

```{r alpha_div_func_plot, echo=TRUE, message=FALSE, warning=FALSE}
alpha_div_func_met<-alpha_div %>%
			select(sample,func) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(metadata, by = join_by(sample == EHI_number))


ggplot(alpha_div_func_met, aes(x = value, y = Elevation)) +
  geom_point() +
  geom_smooth(method = lm) +
  facet_wrap(~ factor(Transect))+
  labs(x = "Alpha div. functional")
```

### Mixed models

#### Neutral diversity

```{r mixed_models_neutral, echo=TRUE, message=FALSE, warning=FALSE}
#Create mixed model formula
formula <- formula(value ~ log(sequencing_depth)+Elevation*Transect+(1|Sampling_point))

#Fit the mixed model
model <- lmer(formula, data = alpha_div_neutral_met)

#Print the model summary
summary(model)
MuMIn::r.squaredGLMM(model)
```

#### Phylogenetic diversity

```{r mixed_models_phylo, echo=TRUE, message=FALSE, warning=FALSE}
#Create mixed model formula
formula <- formula(value ~ log(sequencing_depth)+Elevation*Transect+(1|Sampling_point))

#Fit the mixed model
model <- lmer(formula, data = alpha_div_phylo_met)

#Print the model summary
summary(model)
MuMIn::r.squaredGLMM(model)
```

#### Functional diversities

```{r mixed_models_func, echo=TRUE, message=FALSE, warning=FALSE}
#Create mixed model formula
formula <- formula(value ~ log(sequencing_depth)+Elevation*Transect+(1|Sampling_point))

#Fit the mixed model
model <- lmer(formula, data = alpha_div_func_met)

#Print the model summary
summary(model)
MuMIn::r.squaredGLMM(model)
```


## Beta diversity

```{r beta_colors}
beta_colors <- c("#e5bd5b","#6b7398","#76b183","#d57d2c","#2a2d26","#f9d4cc","#3c634e","#ea68c3")
```

### Beta diversity metrics

```{r beta_diversity, warning=FALSE, comments="", message=FALSE}
beta_q1n <- hillpair(count_table_cov_size,q=1)
#beta_q1p <- hillpair(count_table_cov_size,q=1, tree=tree, metric="S")
beta_q1f <- hillpair(count_table_cov_size_filt,q=1, dist=dist)
```

### Beta diversity plots

#### Neutral
```{r beta_div_neutral_nmds, echo=TRUE, results = FALSE}
beta_q1n_nmds <- beta_q1n$S %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                left_join(metadata, by = join_by(sample == EHI_number))
```

```{r beta_div_neutral_centroids, echo=TRUE}
group_n <- length(unique(beta_q1n_nmds$Transect))
beta_q1n_nmds$Elevation<-as.character(beta_q1n_nmds$Elevation)

beta_q1n_nmds %>%
            group_by(region) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=Elevation, shape=Transect)) +
                #scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Transect"))
```

#### Phylogenetic

```{r beta_div_phylo_nmds, echo=TRUE, results = FALSE}
beta_q1p_nmds <- beta_q1p %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                left_join(sample_table, by = join_by(sample == EHI_number))
```

```{r beta_div_phylo_centroids, echo=TRUE}
group_n <- length(unique(beta_q1p_nmds$Transect))
beta_q1p_nmds$Elevation<-as.character(beta_q1p_nmds$Elevation)

beta_q1p_nmds %>%
            group_by(region) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=Elevation, shape=Transect)) +
                #scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Transect"))
```

#### Functional

```{r beta_div_func_nmds, echo=TRUE, results = FALSE}
beta_q1f_nmds <- beta_q1f$S %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                left_join(metadata, by = join_by(sample == EHI_number))
```

```{r beta_div_func_centroids, echo=TRUE}
group_n <- length(unique(beta_q1f_nmds$Transect))
beta_q1f_nmds$Elevation<-as.character(beta_q1f_nmds$Elevation)


beta_q1f_nmds %>%
            group_by(region) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=Elevation, shape=Transect)) +
                #scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Transect"))
```


### Permanova

#### Neutral

```{r beta_div_neutral_permanova, comment="", echo=TRUE, message=FALSE, warning=FALSE}
sample_table_adonis <- metadata %>%
    filter(EHI_number %in% labels(beta_q1n$S)) %>%
    arrange(EHI_number) %>%
    mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
    select(EHI_number,Transect,Sampling_point,Elevation, sample_code) %>%
    select_if(~ length(unique(.)) > 1) %>% #remove columns with all-identical values
    column_to_rownames(var = "EHI_number") %>%
    as.data.frame()
```

```{r beta_div_neutral_permanova_1, comment="", echo=TRUE, message=FALSE, warning=FALSE}
adonis2(formula=beta_q1n$S ~ Elevation+Transect+Sampling_point, data=sample_table_adonis[labels(beta_q1n$S),], permutations=999) %>%
    as.matrix() %>%
    kable()
```
  
#### Phylogenetic

```{r beta_div_phylo_permanova, comment="", echo=TRUE, message=FALSE, warning=FALSE}
sample_table_adonis_phylo <- metadata %>%
    filter(EHI_number %in% labels(beta_q1p$S)) %>%
    arrange(EHI_number) %>%
    mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
    select(EHI_number,Transect,Sampling_point,Elevation, sample_code) %>%
    select_if(~ length(unique(.)) > 1) %>% #remove columns with all-identical values
    column_to_rownames(var = "EHI_number") %>%
    as.data.frame()
```

```{r beta_div_phylo_permanova_1, comment="", echo=TRUE, message=FALSE, warning=FALSE}
adonis2(formula=beta_q1p$S ~ Elevation+Transect+Sampling_point, data=sample_table_adonis_phylo[labels(beta_q1p$S),], permutations=999) %>%
    as.matrix() %>%
    kable()
```

#### Functional

```{r beta_div_func_permanova, comment="", echo=TRUE, message=FALSE, warning=FALSE}
sample_table_adonis_func <- metadata %>%
    filter(EHI_number %in% labels(beta_q1f$S)) %>%
    arrange(EHI_number) %>%
    mutate(location=paste0(round(longitude,2),"_",round(latitude,2))) %>%
    select(EHI_number,Transect,Sampling_point,Elevation, sample_code) %>%
    select_if(~ length(unique(.)) > 1) %>% #remove columns with all-identical values
    column_to_rownames(var = "EHI_number") %>%
    as.data.frame()
```

```{r beta_div_phylo_permanova_1, comment="", echo=TRUE, message=FALSE, warning=FALSE}
adonis2(formula=beta_q1f$S ~ Elevation+Transect+Sampling_point, data=sample_table_adonis_func[labels(beta_q1f$S),], permutations=999) %>%
    as.matrix() %>%
    kable()
```
