---
title: "Elevation_hologenomcis_func_div_v2"
author: "Garazi Martin Bideguren"
date: "2023-12-13"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Data preparation

### Load required libraries

```{r libraries, message=FALSE, results=FALSE}
library(tidyverse)
library(ape)
library(ggplot2)
library(readxl)
library(writexl)
library(data.table)
library(distillR)
library(RColorBrewer)
library(reshape2)
library(stringr)
library(Rtsne)
library(ggpubr)
library(phyloseq)
library(microbiome)
library(microbiomeutilities)
library(matrixStats)
library(LinDA)
library(SummarizedExperiment)
```

### Declare directories

```{r directories}
workingdir="/Users/dlz554/Desktop/PhD_project/Elevation_Hologenomics/Rmd/Elevatin_Hologenomics_v2"
```

# Diversity analyses

## Functional diversity calculations

```{r functional, message=FALSE, results=FALSE}
#Create merge annotation file
directory_path <- "/Users/dlz554/Desktop/PhD_project/Elevation_Hologenomics/Rmd/Elevation_Hologenomics_v2/DRAM" #Directory of the tsv files (annotations.tsv)
file_list <- list.files(directory_path, pattern = "\\.tsv", full.names = TRUE) #Create a list with all the files
merged_data <- data.frame() #Create an empty data frame
for (file in file_list) {
  data <- read_tsv(file)
  merged_data <- rbind(merged_data, data) #Read each tsv and merge them all together in the data frame
}


#Annotation file
write.table(merged_data,"annotation_file.tsv",row.names=T,col.names=T,quote=F,sep="\t")

#GIFT distillation
GIFTs <- distill(merged_data,GIFT_db,genomecol=2,annotcol=c(9,10,19))
write.table(GIFTs,"GIFTs.tsv",row.names=T,col.names=T,quote=F,sep="\t")

#Merge GIFTs into higher hierarchical levels
GIFTs_elements <- to.elements(kegg_table,GIFT_db)
GIFTs_functions <- to.functions(GIFTs_elements,GIFT_db)
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

# GIFT statistics
#Max GIFT values
GIFTs_elements %>%
  as.data.frame() %>%
  summarise_if(is.numeric, max)
```

```{r GIFT_heatmap, message=FALSE, results=FALSE}
#Order by tips
tiporder <- function(phy, labels = TRUE){
  nn <- length(phy$tip.label) #How many tips on the tree?
  edge <- phy$edge
  nums <- rev(edge[edge[,2] %in% 1:nn, 2])
  if(labels == TRUE) phy$tip.label[nums] else nums
}
```

```{r GIFT_heatmap_plot, message=FALSE, results=TRUE}
#Plot heatmap (genomes sorted by phylogeny)
GIFTs_elements %>%
  reshape2::melt() %>%
  rename(Genome = Var1, Code_element = Var2, GIFT = value) %>%
  inner_join(GIFT_db,by="Code_element") %>%
  mutate(Genome = factor(Genome, levels = tiporder(tree))) %>%
  ggplot(., aes(x=Code_element, y=Genome, fill=GIFT, group=Code_function))+
  geom_tile()+
  scale_y_discrete(guide = guide_axis(check.overlap = TRUE))+
  scale_x_discrete(guide = guide_axis(check.overlap = TRUE))+
  scale_fill_gradientn(limits = c(0,1), colours=brewer.pal(7, "YlGnBu"))+
  facet_grid(. ~ Code_function, scales = "free", space = "free")+
  theme_grey(base_size=8)+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```

```{r GIFT_heatmap_plot_1, message=FALSE, results=TRUE}
#Plot heatmap (faceted by coassembly)
GIFTs_elements %>%
  reshape2::melt() %>%
  rename(Genome = Var1, Code_element = Var2, GIFT = value) %>%
  inner_join(GIFT_db,by="Code_element") %>%
  mutate(Genome = factor(Genome, levels = tiporder(tree))) %>%
  mutate(Coassembly = Genome) %>%
  mutate_at(vars(Coassembly), ~ str_replace(., "_bin.*", "")) %>%
  ggplot(., aes(x=Code_element, y=Genome, fill=GIFT, group=Code_function))+
  geom_tile()+
  scale_y_discrete(guide = guide_axis(check.overlap = TRUE))+
  scale_x_discrete(guide = guide_axis(check.overlap = TRUE))+
  scale_fill_gradientn(limits = c(0,1), colours=brewer.pal(7, "YlGnBu"))+
  facet_grid(Coassembly ~ Code_function, scales = "free", space = "free")+
  theme_grey(base_size=8)+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```

```{r GIFT_tSNE, echo=TRUE, results=FALSE}
# Taxonomy table preparation
summary_file <- read.table("DMB0113_gtdbtk_combined_summary.tsv",sep="\t",header=T)
summary_file$classification

# Split name column into firstname and last name
summary_file %>%
  select(user_genome, classification) %>%
  rename(Genome = user_genome) %>%
  separate(classification, c('Domain', 'Phylum','Class','Order','Family','Genus','Species'),sep=";") %>%
  mutate_at(vars(Domain,Phylum,Class,Order,Family,Genus,Species), ~ str_replace(., "[dpcofgs]__", "")) %>%
  mutate_at(vars(Genome), ~ str_replace(., ".fa", "")) %>%
  write.table(.,"taxonomy.tsv",col.names=T,row.names=F,sep="\t",quote=F)

# GIFT ordination
taxonomy <- read.table("taxonomy.tsv",sep="\t",header=T)

set.seed(100)
GIFTs_elements_tSNE <- Rtsne(X=GIFTs_elements, dims = 2, check_duplicates = FALSE)

GIFTs_domains <- as.data.frame(GIFTs_domains)
GIFTs_domains$Genome <- rownames(GIFTs_domains)

GIFTs_elements_tSNE <- GIFTs_elements_tSNE$Y %>%
  as.data.frame() %>%
  mutate(Genome=rownames(GIFTs_elements)) %>%
  inner_join(GIFTs_domains, by="Genome") %>%
  inner_join(taxonomy, by="Genome") %>%
  mutate_at(vars(Phylum, Class, Order, Family, Genus), factor) %>%
  #mutate(cyl = factor(Phylum, levels = phylum_colors$Phylum)) %>%
  rename(tSNE1="V1", tSNE2="V2")
```

### Ordination of Phylum level MAGs functional diversity

```{r GIFT_tSNE_plot, echo=TRUE, results=TRUE}
GIFTs_elements_tSNE %>%
  ggplot(aes(x = tSNE1, y = tSNE2, color = Phylum))+
  geom_point(size=1, shape=16, alpha=0.8) +
  #scale_color_gradientn(limits = c(0,1), colours=brewer.pal(7, "YlGnBu"))+
  theme_minimal() +
  theme()
```

### Ordination of overall MAGs functional diversity

```{r GIFT_tSNE_plot_1, echo=TRUE, results=TRUE}
GIFTs_elements_tSNE %>%
  ggplot(aes(x = tSNE1, y = tSNE2, color = Overall))+ #coloured by GIFT
  geom_point(size=1, shape=16, alpha=0.8) +
  scale_color_gradientn(limits = c(0,1), colours=brewer.pal(7, "YlGnBu"))+
  theme_minimal() +
  theme()
```

### Functional diversity plots

```{r GIFT_community, echo=TRUE, results=TRUE}
#Get community-weighed average GIFTs per sample. Remember to use the count table already filtered by the MAGs that are present in the kegg_table (GIFTs)
count_filtered_func <- count_table_cov_size[rownames(count_table_cov) %in% rownames(kegg_table),]
GIFTs_elements_community <- to.community(GIFTs_elements,count_filtered_func,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,count_filtered_func,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,count_filtered_func,GIFT_db)

#Merge the functional domains with the metadata
merge_gift <- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "EHI_number") %>% 
  merge(., metadata, by="EHI_number")
merge_gift$Elevation<-as.factor(merge_gift$Elevation)
```

#### Comparison by elevation
##### Biosynthesis

```{r biosynthesis_plot, echo=TRUE, results=TRUE}
merge_gift %>%
  ggplot(aes(x=Elevation,y=Biosynthesis,color=Elevation,fill=Elevation))+
  geom_jitter(width = 0.2, size = 3, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  #scale_color_manual(values=alpha_colors[c(1:group_n)]) +
  #scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)],"50")) +
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10), #, size=18
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        #         axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        #          axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        #          panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Elevation (m)")
```

##### Degradation

```{r degradation_plot, echo=TRUE, results=TRUE}
merge_gift %>%
  ggplot(aes(x=Elevation,y=Degradation,color=Elevation,fill=Elevation))+
  geom_jitter(width = 0.2, size = 3, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  #scale_color_manual(values=alpha_colors[c(1:group_n)]) +
  #scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)],"50")) +
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10), #, size=18
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        #         axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        #         axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        #          panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Elevation (m)")
```

##### Structure

```{r structure_plot, echo=TRUE, results=TRUE}
 merge_gift %>%
  ggplot(aes(x=Elevation,y=Structure,color=Elevation,fill=Elevation))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  #scale_color_manual(values=alpha_colors[c(1:group_n)]) +
  #scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)],"50")) +
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10), #, size=18
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        #          axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        #          axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        #          panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Elevation (m)")
```

##### Overall

```{r overall_plot, echo=TRUE, results=TRUE}
merge_gift %>%
  ggplot(aes(x=Elevation,y=Overall,color=Elevation,fill=Elevation))+
  geom_jitter(width = 0.2, size = 3, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  #scale_color_manual(values=alpha_colors[c(1:group_n)]) +
  #scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)],"50")) +
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10), #, size=18
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        #          axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        #          axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        #          panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Elevation (m)")
```

#### Comparison by transect
##### Biosynthesis

```{r biosynthesis_plot_1, echo=TRUE, results=TRUE}
merge_gift %>%
  ggplot(aes(x=Transect,y=Biosynthesis,color=Transect,fill=Transect))+
  geom_jitter(width = 0.2, size = 3, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  scale_color_manual(values=alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)],"50")) +
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10), #, size=18
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        #         axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        #          axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        #          panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Transect")
```

##### Degradation

```{r degradation_plot_1, echo=TRUE, results=TRUE}
merge_gift %>%
  ggplot(aes(x=Transect,y=Degradation,color=Transect,fill=Transect))+
  geom_jitter(width = 0.2, size = 3, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  scale_color_manual(values=alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)],"50")) +
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10), #, size=18
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        #         axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        #         axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        #          panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Transect")
```

##### Structure

```{r structure_plot, echo=TRUE, results=TRUE}
 merge_gift %>%
  ggplot(aes(x=Transect,y=Structure,color=Transect,fill=Transect))+
  geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  scale_color_manual(values=alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)],"50")) +
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 3, size=10), #, size=18
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        #          axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        #          axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        #          panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Transect")
```

##### Overall

```{r overall_plot_1, echo=TRUE, results=TRUE}
merge_gift %>%
  ggplot(aes(x=Transect,y=Overall,color=Transect,fill=Transect))+
  geom_jitter(width = 0.2, size = 3, show.legend = FALSE)+ 
  geom_boxplot(alpha=0.2,outlier.shape = NA, width = 0.5, show.legend = FALSE, coef=0)+
  scale_color_manual(values=alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values=paste0(alpha_colors[c(1:group_n)],"50")) +
  stat_compare_means() +
  theme(axis.text.x = element_text(vjust = 0.5, size=10), #, size=18
        axis.text.y = element_text(size=10),
        axis.title=element_text(size=12,face="bold"),
        axis.text = element_text(face="bold", size=18),
        #          axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        #          axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        #          panel.border = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        legend.text = element_text(size=10),
        legend.title = element_text(size=12),
        legend.position="none",
        legend.key.size = unit(1, 'cm'),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))+
  labs( x = "Transect")
```

## Functional diversity statistics



